compiler/src/dmd/func.d

```diff
@@ -2199,6 +2199,19 @@ extern (C++) class FuncDeclaration : Declaration
         }
         else
         {
+            foreach (v; closureVars)
+            {
+                v.doNotInferScope = true;
+                v.storage_class &= ~STC.maybescope;
+                if ((v.isScope() || (v.storage_class & STC.scopeinferred)) &&
+                    (global.params.useDIP1000 == FeatureState.enabled && isSafe()))
```

Comment:
Don't introduce branches with `global.params.useDIP1000 == FeatureState.enabled`, `-preview=dip1000` is now enabled by default in deprecation mode so use `setUnsafeDIP1000` and/or `previewErrorFunc`.

Reply:
setUnsafeDIP1000() takes `sc` as an argument, which is not available in this function.